src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -362,6 +362,8 @@ private void ProcessSetting(long id, long value)
     private ValueTask ProcessGoAwayFrameAsync()
     {
         EnsureSettingsFrame(Http3FrameType.GoAway);
+        _context.Connection.StopProcessingNextRequest(serverInitiated: false);
```

Comment:
```suggestion
        // StopProcessingNextRequest must be called before RequestClose to ensure it's considered client initiated.
        _context.Connection.StopProcessingNextRequest(serverInitiated: false);
```

Also, not related to this PR exactly, but the two StopProcessingNextRequest methods in Http3ControlStream itself are not called by anything, right? At first, I thought you were calling one of those, and it seemed odd. Can we delete them?

Reply:
Sure, I don't see those being used.